### PR TITLE
feat: Add gtm

### PR DESCRIPTION
**Pull request by sdmcraft**
Test URLs:
- Before: https://main--icicidirect--aemsites.hlx.live/research/equity
- After: https://gtm--icicidirect--aemsites.hlx.live/research/equity

- [ ] New Blocks introduced in this PR
      If yes, please provide details below

Block Name    | Documentation
------------- | -------------
 For e.g. carousel | [doc-link](https://main--icicidirect--aemsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/carousel&index=0)

### CODE REVIEW

**Comment by anagarwa** [comment]
This covers below third Party calls. 
1. Google Analytics/ Tag Manager
2. Quora Ads
3.  LinkedInAds
4. Twitter
5. Hotjar

Not Covered. 
Bing Ads
Cloudflare Captcha which seems to be cover by https://github.com/aemsites/icicidirect/pull/80

**Review by JiangLong2019 — APPROVED** [review]

File: scripts/delayed.js [hunk 1]
```diff
@@ -5,3 +5,28 @@ import { loadScript, sampleRUM } from './aem.js';
...
+  const dl = l !== 'dataLayer' ? `&l=${l}` : '';
+  j.async = true;
+  j.src = `https://www.googletagmanager.com/gtm.js?id=${i}${dl}`;
+  f.parentNode.insertBefore(j, f);
+}(window, document, 'script', 'dataLayer', 'GTM-WF9LTLZ'));
+
+(function loadTagManagerElement() {
+  const htmlContent = '<iframe src="https://www.googletagmanager.com/ns.html?id=GTM-WF9LTLZ"\n'
+      + 'height="0" width="0" style="display:none;visibility:hidden"></iframe>';
+  const scriptElement = document.createElement('noscript');
```
Comment (sdmcraft) [inline scripts/delayed.js:28]: We don't need the `noscript` as in that case this script itself will not run :)

File: scripts/cookie-script.js [hunk 2]
```diff
@@ -0,0 +1,328 @@
+/* eslint-disable */
```
Comment (sdmcraft) [inline scripts/cookie-script.js:1]: May I suggest to add the cookie script in a separate PR and keep this PR limited to GTM.

File: scripts/delayed.js (hunk 1 above)
Comment (anagarwa) [reply scripts/delayed.js:28]: @sdmcraft  They have the same way in their current website as well and this is what that has been shared too. Also as per [0], `noscript` tag is supposed to work when browser does not support script or scripts has been disabled at browser. Ami I missing something here or got the interpretation of the `noscript` tag wrong ? 

[0]https://developer.mozilla.org/en-US/docs/Web/HTML/Element/noscript

File: scripts/cookie-script.js (hunk 2 above)
Comment (anagarwa) [reply scripts/cookie-script.js:1]: @sdmcraft  sure, let me update the pr

**Comment by anagarwa** [comment]
Regarding the failing of psi check here, when i run the page in chrome browser it gives me below details which signifies the need to run in incognito.
image

and Incognito modes provides the perfect score for the page
image

as well the pagespeed check
image

File: scripts/delayed.js (hunk 1 above)
Comment (sdmcraft) [reply scripts/delayed.js:26]: if the browser does not support javascript, then `delayed.js` where you are proposing this change would not run.

Comment (anagarwa) [reply scripts/delayed.js]: thats true, missed this

**Review by anagarwa — APPROVED** [review]

**Pull request merged**